templates/project/.github/workflows/test.yaml.twig

```diff
@@ -0,0 +1,118 @@
+          - php-version: '{{ target_php|default(php|last) }}'
+            dependencies: highest
+            allowed_to_fail: true
+            allow_deprecations: false
+            variant: normal
+          - php-version: '8.0'
+            dependencies: highest
+            allowed_to_fail: true
+            allow_deprecations: true
+            variant: normal
+{% for package_name, package_versions in versions %}
+{% for version in package_versions %}
+          - php-version: '{{ target_php|default(php|last) }}'
+            dependencies: highest
+            allowed_to_fail: false
+            allow_deprecations: true
+            variant: '{{ packages[package_name] }}:"{{ version }}.*"'
+{% endfor %}
+{% if package_versions|length > 0 and not is_dev_master(package_versions|last) %}
+          - php-version: '{{ target_php|default(php|last) }}'
+            dependencies: highest
+            allowed_to_fail: true
+            allow_deprecations: true
+            variant: '{{ packages[package_name] }}:"dev-master as {{ package_versions|last }}.x-dev"'
+{% endif %}
+{% endfor %}
+{% verbatim %}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          coverage: pcov
+          tools: composer:v2
+
+      - name: Add PHPUnit matcher
+        run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Set Composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache Composer
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-${{ matrix.variant }}
+          restore-keys: ${{ runner.os }}-${{ matrix.php-version }}-composer-
+
+      - name: Set Environment variable
+        if: ${{ !matrix.allow_deprecations }}
+        run: echo "::set-env name=SYMFONY_DEPRECATIONS_HELPER::0"
+
+      - name: Install variant
+        if: matrix.variant != 'normal'
+        run: composer require ${{ matrix.variant }} --no-update
+
+      - name: Install Composer dependencies (lowest)
+        if: matrix.dependencies == 'lowest'
+        run: composer update --prefer-dist --no-progress --no-interaction --prefer-stable --prefer-lowest
+
+      - name: Install Composer dependencies (highest)
+        if: matrix.dependencies == 'highest' && matrix.php-version != '8.0'
+        run: composer update --prefer-dist --no-progress --no-interaction --prefer-stable
+
+      - name: Install Composer dependencies (highest) on PHP 8.0
+        if: matrix.dependencies == 'highest' && matrix.php-version == '8.0'
+        run: composer update --prefer-dist --no-progress --no-interaction --prefer-stable --ignore-platform-req=php
```

Comment:
This one is an improvement over travis. Because we are using composer 2, we can ignore php platform req of other packages, so we can test with php 8. There is still a failure, because we are using symfony phpunit bridge with phpunit 8.5, and that is not compatible with php 8. (we should move to phpunit 9.3 that build, or all the builds if we can)